authors/apps/authentication/backends.py

```diff
@@ -1,10 +1,69 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+import jwt
 
-"""Configure JWT Here"""
+from django.conf import settings
 
+from rest_framework import authentication, exceptions
+
+from .models import User
+
+
+class JWTAuthentication(authentication.BaseAuthentication):
+    authentication_header_prefix = 'Token'
+
+    def authenticate(self, request):
+        """
+        This method is called for on every request and it
+        returns a two-tuple of (user, token) if authentication is
+        successfull and a None otherwise.
+        If any fails we raise `AuthenticationFailed` and Django rest framework
+        does the rest.
+        """
+        request.user = None
+
+        authentication_header = authentication.get_authorization_header(request).split()
+        header_prefix = self.authentication_header_prefix.lower()
+
+        if not authentication_header:
+            #NO authentication header value was entered
+            return None
+
+        if len(authentication_header) == 1:
+            # Either token or prefix are missing or 
+            # Token is either connected with the prefix.
+            return None
+
+        if len(authentication_header)>2:
+            #The header entered has more than one space
+            return None
+        
+        #decoding the prefix and toke from byte format to a format that can be 
+        #easily handled
+        prefix = authentication_header[0].decode('utf-8')
+        token = authentication_header[1].decode('utf-8')
+
+        if prefix.lower() != header_prefix:
+            #the decoded prefix is not equal to the authentication header prefix
+            return None
+        
+        #if all the above is passed, we then go one to authenticate
+        #the given credentials. 
+        return self.authenticate_user_details(request, token)
+
+    def authenticate_user_details(self, request, token):
+        try:
+            payload = jwt.decode(token, settings.SECRET_KEY)
+        except:
+            raise exceptions.AuthenticationFailed('The authentication is invalid')
```

Comment:
I am guessing this should be `token is invalid`

Reply:
Have done it.